### PR TITLE
feat(web, cli): sync the url between http server and dashboard

### PR DESCRIPTION
**Pull request by djabarovgeorge**
#### What?

Currently all navigation in Local Studio is performed under a single location as far as the user can see.

We need to add a mini-router to the Local Studio to ensure that the iFramed application communicates to the host (proxy) application to update the page location via history API.

Reloading a page should respect the current URL in the address bar, loading into the correct page of the application

####  Why? (Context)

We need to ensure that movement across the page and reloads to the page are respected.

####  Definition of Done
* Moving around the application in Local Studio results in updates to the URL
* Refreshing the page when on a custom route reloads the page with the expected content for that route
### Screenshots
https://www.loom.com/share/3cbfd73e762e48c287d69646afa545f1

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/web/src/hooks/useAuth.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export function useAuth() {
   const queryClient = useQueryClient();
   const navigate = useNavigate();
   const location = useLocation();
-  const inPublicRoute = Array.from(PUBLIC_ROUTES_PREFIXES.values()).find((prefix) =>
+  const inPublicRoute = !!Array.from(PUBLIC_ROUTES_PREFIXES.values()).find((prefix) =>
```
Comment (djabarovgeorge) [inline apps/web/src/hooks/useAuth.ts]: was `string|undefined` before and not boolean.

File: apps/web/src/components/execution-detail/ExecutionDetailsStepHeader.tsx [hunk 2]
```diff
@@ -100,7 +100,7 @@ const generateDetailByStepAndStatus = (status, job) => {
     const { digest, step: stepMetadata, payload } = job;
 
     if (!digest.amount && !digest.unit) return `Waiting to receive execution delay from bridge endpoint`;
-    if (stepMetadata.metadata.type === DelayTypeEnum.SCHEDULED) {
+    if (stepMetadata?.metadata?.type === DelayTypeEnum.SCHEDULED) {
```
Comment (rifont) [inline apps/web/src/components/execution-detail/ExecutionDetailsStepHeader.tsx:103]: This is a patch for the following error:
```
TypeError: Cannot read properties of undefined (reading 'type')
    at generateDetailByStepAndStatus (ExecutionDetailsStepHeader.tsx:103:1)
    at ExecutionDetailsStepHeader (ExecutionDetailsStepHeader.tsx:140:1)
    at renderWithHooks (react-dom.development.js:15486:1)
    at updateFunctionComponent (react-dom.development.js:19617:1)
    at beginWork (react-dom.development.js:21640:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)
    at beginWork$1 (react-dom.development.js:27490:1)
    at performUnitOfWork (react-dom.development.js:26596:1)
```

File: apps/web/src/components/execution-detail/helpers.ts [hunk 3]
```diff
@@ -78,6 +79,10 @@ export const getLogoByType = (
     return Mobile;
   }
 
+  if (type === StepTypeEnum.CUSTOM) {
+    return CurlyBrackets;
+  }
+
```
Comment (rifont) [inline apps/web/src/components/execution-detail/helpers.ts:85]: Adding the missing icon for custom

image

**Review by rifont — APPROVED** [review]
Awesome work! Super clean

**Pull request merged**